### DISCUSSION TITLE
How do you manage to get Scala 3 code snippets in NVim?

### DISCUSSION THREAD

**Discussion by cruiztorresj** [scalameta/nvim-metals#684] [Q&A]
Hello there all,

I've followed the example configuration you guys have over [here](https://github.com/scalameta/nvim-metals/discussions/39), I am not sure if I did it right however. Because I am only getting Scala 2 syntax code snippets.

This is the [configuration ](https://github.com/cruiztorresj/MyNvimConfiguration) I have in place.

Ideally, I would like to have the kind of code completion [Scastie](https://scastie.scala-lang.org) offers you. It even detects the exhaustive cases when you are performing a match.

I am using `luasnip` instead, and drawing snippets from this [plugin](https://github.com/rafamadriz/friendly-snippets). This is the section related to [Scala](https://github.com/rafamadriz/friendly-snippets/blob/main/snippets/scala.json), and as you can see, all snippets being offered are related to Scala 2.

I am not trying to promote myself but I streamed the whole process because I thought someone else may benefit from it. Here's a [video ](https://www.youtube.com/watch?v=SZ5yg7iyvho) of the above paragraphs.

So, how did you guys did it? Are you getting Scala 3 code completion assistance within NVim?

Thanks a lot!

**Comment by ckipp01** [comment]
Hey @cruiztorresj, yes I do get snippets for Scala 3


https://github.com/user-attachments/assets/d18c241a-13dd-48cd-bf94-e01baf3b3707

So I'm not 100% sure what's going wrong with your situation. Does your setup work for snippets in other languages or versions of Scala?

**Reply by cruiztorresj** [reply]
I got it working some weeks ago.
Thanks a lot for your answer.
